packages/TimePicker/src/TimePicker.js

```diff
@@ -137,94 +186,72 @@ export default class TimeInput extends React.Component {
       return;
     }
 
-    this.setState({
-      isTabIndexActive: 0,
-      isVisible: false,
-    });
+    setIsTabIndexActive(0);
+    setIsOpen(false);
   };
 
-  handleClickOutside = () => {
-    this.finish();
-  };
+  function finish() {
+    setIsOpen(false);
+    setValue(timeStr);
+  }
 
-  handleKeyUp = event => {
+  const handleKeyUp = event => {
     if (event.key === "Enter") {
-      this.finish();
+      finish();
     }
 
     if (event.key === "Escape") {
-      this.finish();
+      finish();
     }
   };
 
-  handleError(time) {
-    this.setState({
-      ...time,
-      error: this.props.labelError,
-      isVisible: false,
-    });
-  }
-
-  finish() {
-    this.setState(state => ({
-      isVisible: false,
-      value: state.timeStr,
-    }));
-  }
-
-  render() {
-    const {
-      isDisabled,
-      ariaLabel,
-      labelCustom,
-      labelHours,
-      labelMinutes,
-      labelPeriod,
-      labelAM,
-      labelPM,
-      ...moreProps
-    } = this.props;
-
-    return (
-      <sc.CSSHolder>
-        <scPicker.CSSHolder>
-          <div className="timeinput" onFocus={this.handleFocus} onBlur={this.handleBlur}>
-            {/* eslint-disable-next-line */}
-            <div tabIndex={this.state.isTabIndexActive ? 0 : -1}>
-              {/* setting hasClearButton to false confuses, look like a close button for the timeinput hasClearButton */}
-              <div ref={this.$refInputContainer}>
-                <Input
-                  ariaLabel={ariaLabel}
-                  hasClearButton={false}
-                  isDisabled={isDisabled}
-                  onInput={this.handleChange}
-                  onFocus={this.handleFocus}
-                  onKeyUp={this.handleKeyUp}
-                  value={this.state.value}
-                  data-qa-id="time-input__starting-at"
-                  {...moreProps}
-                />
-              </div>
-              {/** THIS ERROR SHOULD BE REPLACE WITH A PROPER FORM-ELEMENT */}
-              <span>{this.state.error}</span>
-              <Picker
-                hh={this.state.hh}
-                isVisible={this.state.isVisible}
-                labelCustom={labelCustom}
-                labelHours={labelHours}
-                labelMinutes={labelMinutes}
-                labelPeriod={labelPeriod}
-                labelAM={labelAM}
-                labelPM={labelPM}
-                mm={this.state.mm}
-                onClick={this.handleClick}
-                period={this.state.period}
+  return (
+    <L10n>
+      <sc.TimePicker onFocus={handleFocus} onBlur={handleBlur}>
+        <Popover style={{ width: "100%" }} isOpen={isOpen} edge="left" offset={0} align="bottom">
+          {/* eslint-disable-next-line */}
+          <div tabIndex={isTabIndexActive ? 0 : -1}>
+            {/* setting hasClearButton to false confuses, look like a close button for the timeinput hasClearButton */}
```

Comment:
@nahumzs do you want to address this comment?

Reply:
ohh that was because we were using the clear button on the `<Input hasClearButton />` but looked like if you were having a close button instead of a clear one, and the user might confuse to click it to close the popover but instead clear the value. 

I think we can remove the comment.